test/test_util.py

```diff
@@ -280,3 +280,33 @@ def test_detect_running_process_id(self, cmd_str):
             raise AssertionError(f"Could not match command or matched twice:\n{output_bytes.decode('ascii')}")
         no_id, output_bytes = myhoard_util.detect_running_process_id("certainlynosuchprocesscurrentlyrunning")
         assert no_id is None
+
+
+def test_restart_unexpected_dead_sql_thread():
```

Comment:
nit. Add return type of `-> None`, our mypy config isn't great yet, but this will help detect issues as we increase the scope of mypy checks.

Reply:
Done.